src/seedsigner/views/psbt_views.py

```diff
@@ -128,22 +131,27 @@ def run(self):
             else:
                 num_self_transfer_outputs += 1
 
-        screen = psbt_screens.PSBTOverviewScreen(
-            spend_amount=psbt_parser.spend_amount,
-            change_amount=psbt_parser.change_amount,
-            fee_amount=psbt_parser.fee_amount,
-            num_inputs=psbt_parser.num_inputs,
-            num_self_transfer_outputs=num_self_transfer_outputs,
-            num_change_outputs=num_change_outputs,
-            destination_addresses=psbt_parser.destination_addresses,
-        )
+        spend_amount = psbt_parser.spend_amount
+        change_amount = psbt_parser.change_amount
+        fee_amount = psbt_parser.fee_amount
+        num_inputs = psbt_parser.num_inputs
+        destination_addresses = psbt_parser.destination_addresses
 
         # Everything is set. Stop the loading screen
         if self.loading_screen:
             self.loading_screen.stop()
 
         # Run the overview screen
-        selected_menu_num = screen.display()
+        selected_menu_num = self.run_screen(
+            PSBTOverviewScreen,
+            spend_amount=spend_amount,
+            change_amount=change_amount,
+            fee_amount=fee_amount,
+            num_inputs=num_inputs,
+            num_self_transfer_outputs=num_self_transfer_outputs,
+            num_change_outputs=num_change_outputs,
+            destination_addresses=destination_addresses
+        )
```

Comment:
same comments as Keith's (lines 134-138 perhaps unneeded as locals, could be passed on in the run_screen() call).